pkg/server/plugin/nodeattestor/awsiid/iid.go

```diff
@@ -223,14 +227,10 @@ func (p *IIDAttestorPlugin) Configure(ctx context.Context, req *configv1.Configu
 		return nil, status.Errorf(codes.InvalidArgument, "unable to decode configuration: %v", err)
 	}
 
-	// Get the AWS CA public key. We do this lazily on configure so deployments
+	// Function to get the AWS CA certificate. We do this lazily on configure so deployments
```

Comment:
Did you measure the startup cost if we parse all of the certificates in advance? I'd be surprised if the impact was substantial. I'd also much prefer to pay that cost once instead of every time we attest a new node. My vote would be to either 1) just parse them all on startup, or 2) cache them once parsed so we don't pay the cost continuously.

Reply:
I prefer option 2. Rather than parsing all 45 certificates on startup, I'd choose parsing and caching the certificates on demand. This way, only the certificates that are actually used will be parsed and cached, avoiding unnecessary overhead.